scaladoc/src/dotty/tools/scaladoc/SocialLinks.scala

```diff
@@ -4,11 +4,12 @@ import java.nio.file.Path
 import java.nio.file.Paths
 import dotty.tools.dotc.core.Contexts.Context
 
-enum SocialLinks(val url: String, val className: String):
-  case Github(ghUrl: String) extends SocialLinks(ghUrl, "gh")
-  case Twitter(tUrl: String) extends SocialLinks(tUrl, "twitter")
-  case Gitter(gUrl: String) extends SocialLinks(gUrl, "gitter")
-  case Discord(dUrl: String) extends SocialLinks(dUrl, "discord")
+enum SocialLinks(val url: String, val whiteIcon: String, val darkIcon: String, val className: String):
+  case Github(ghUrl: String) extends SocialLinks(ghUrl, "", "", "gh")
+  case Twitter(tUrl: String) extends SocialLinks(tUrl, "", "", "twitter")
+  case Gitter(gUrl: String) extends SocialLinks(gUrl, "", "", "gitter")
+  case Discord(dUrl: String) extends SocialLinks(dUrl, "", "", "discord")
+  case Custom(cUrl: String, firstIcon: String, secondIcon: String) extends SocialLinks(cUrl, firstIcon, secondIcon, "custom")
```

Comment:
I'm not a fan of adding those fields to the parent class, just so they will be used in one case and empty in all others. I'll suggest something like this:

```suggestion
enum SocialLinks(val url: String, val className: String):
  case Github(ghUrl: String) extends SocialLinks(ghUrl, "gh")
  case Twitter(tUrl: String) extends SocialLinks(tUrl, "twitter")
  case Gitter(gUrl: String) extends SocialLinks(gUrl, "gitter")
  case Discord(dUrl: String) extends SocialLinks(dUrl, "discord")
  case Custom(cUrl: String, lightIcon: String, darkIcon: String) extends SocialLinks(cUrl, "custom")
```

Then you can match on the link in `HtmRenderer.scala`, like so:

```scala
        link match
          case SocialLinks.Custom(_, lightIcon, darkIcon) =>
            Seq(
              img(cls := s"icon-button ${icon(link)}", src := s"../../../../images/$lightIcon"),
              img(cls := s"icon-button ${icon(link)}-dark", src := s"../../../images/$darkIcon")
            )
          case _ =>
            button(cls := s"icon-button ${icon(link)}")
```

Also, it should be `lightIcon`, not `whiteIcon`.